lib/pdc/resource/associations/builder.rb

```diff
@@ -0,0 +1,44 @@
+require 'active_support/dependencies'
+
+module PDC::Resource
+  module Associations
+    class Builder
+      def initialize(parent_class, name, type, options = {})
+        @parent_class = parent_class
+        @name = name
+        @type = type
+        @options = options
+      end
+
+      def build(parent)
+        @type.new(klass, parent, @name, @options)
+      end
+
+      def klass
+        @klass ||= custom_class || compute_class(@name)
+      end
+
+      private
+
+      def custom_class
+        @options[:class_name].constantize if @options[:class_name]
+      end
+
+      # https://github.com/rails/rails/blob/70ac072976c8cc6f013f0df3777e54ccae3f4f8c/activerecord/lib/active_record/inheritance.rb#L132-L150
```

Comment:
Ok. I think it is imitating rails's implementation to compute the class. I copied this from [https://github.com/balvig/spyke](spyke). Actually most of the codes are imitating spyke and do some tweaks for our use case. Ok. I will add some comments here :)

Reply:
Hm... why are we not actually using spyke? Is there something we do different in this gem? I checked the README, but couldn't find an explanation there.

Reply:
@romanofski spyke is a generic gem to Interact with REST services in an ActiveRecord-like manner. It requires activesupport 4.0, but pdc gem needs to run against activesupport 3.2.22. I guess it is because Errata use this version. Also pdc gem is specific for PDC server, we only need to support Get request at least for now, we don't need to support POST/PUT, etc. For example, this association implementation only supports get, no post action.